inductiva/storage/storage.py

```diff
@@ -118,3 +129,154 @@ def _print_contents_table(contents):
         formatters=formatters,
         header_formatters=header_formatters,
     )
+
+
+def upload_from_url(
+    url: str,
+    remote_dir: str,
+    file_name: str = None,
+    unzip: bool = False,
+    overwrite: bool = True,
+):
+    """Upload a file from a URL to the user workspace."""
+    api_instance = storage_api.StorageApi(inductiva.api.get_client())
+
+    if file_name is None:
+        parsed_url = urlparse(url)
+        file_name = unquote(parsed_url.path.split("/")[-1])
+    contents = api_instance.upload_from_url(
+        query_params={
+            "url": url,
+            "file_name": file_name,
+            "unzip": "t" if unzip else "f",
+            "overwrite": "t" if overwrite else "f",
+        },
+        path_params={
+            "folder_name": remote_dir,
+        },
+    )
+    if contents.response.status == 200:
+        logging.info("File is being uploaded...")
+    else:
+        logging.info("File upload failed.")
+
+
+def upload(
+    local_path: str,
+    remote_dir: str = "default",
+    overwrite: bool = False,
+):
+    """Upload local files to the user workspace."""
+    api_instance = storage_api.StorageApi(inductiva.api.get_client())
+
+    input_zip_path = _zip_file_or_folder(local_path)
+
+    zip_file_size = os.path.getsize(input_zip_path)
+    logging.info("Input archive size: %s",
+                 format_utils.bytes_formatter(zip_file_size))
+
+    logging.info("Uploading input...")
+    try:
+
+        api_response = api_instance.get_upload_url(
```

Comment:
If this is the only thing that changes from `upload_input` we can create a method that gets the endpoint as an argument and reuse just passing a new endpoint.

Reply:
I refactored with a new method that accepts the endpoints and is reused by tasks and workspaces